src/include/mpir_datatype.h

```diff
@@ -453,6 +458,23 @@ static inline int MPIR_Datatype_set_contents(MPIR_Datatype * new_dtp,
     return MPI_SUCCESS;
 }
 
+static inline void MPIR_Datatype_get_flattened(MPI_Datatype type, void **flattened,
+                                               int *flattened_sz)
+{
+    MPIR_Datatype *dt_ptr;
+
+    MPIR_Datatype_get_ptr(type, dt_ptr);
+    if (dt_ptr->flattened == NULL) {
+        MPIR_Typerep_flatten_size(dt_ptr, &dt_ptr->flattened_sz);
+        dt_ptr->flattened = MPL_malloc(dt_ptr->flattened_sz, MPL_MEM_DATATYPE);
+        MPIR_Assert(dt_ptr->flattened);
+        MPIR_Typerep_flatten(dt_ptr, dt_ptr->flattened);
+    }
+
+    *flattened = dt_ptr->flattened;
+    *flattened_sz = dt_ptr->flattened_sz;
+}
```

Comment:
Can we avoid `static inline` for this?  It's not a performance critical function.

Reply:
Sure, will move to regular function.